### PR TITLE
chore(web): add snapshot tests

### LINKED ISSUE #709: JestTypeScriptOptions causes a circular dependency through NodePackage by pulling TS into the JS Chain
This later prevents testing of web components like react and next.js

Discussion from @pgollucci  and @eladb  from CDK.dev Slack #projen

https://cdk-dev.slack.com/archives/C01BP9PQHPC/p1618851148176300

This blocks https://github.com/projen/projen/issues/708

### PR DESCRIPTION
**Pull request by Chriscbr**
Fixes #709.

Not detailed tests, but I found it useful to compare the snapshots between the nextjs and nextjs-ts projects (and same goes for react).

Needed to extract `TypescriptConfig` from `typescript.ts` in order to fix a runtime dependency issue, caused by the dependency cycle typescript.ts (TypeScriptAppProject) > node-project.ts (NodeProject) > jest.ts (Jest) > typescript.ts (TypescriptConfig).

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/__tests__/web/nextjs-project.test.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class TestNextJsProject extends NextJsProject {
+  constructor(options: Partial<NextJsProjectOptions> = { }) {
+    super({
+      ...options,
+      clobber: false,
+      name: 'test-nextjs-project',
+      outdir: mkdtemp(),
+      logging: { level: LogLevel.OFF },
+      jsiiFqn: 'projen.web.NextJsProject',
```
Comment (Chriscbr) [inline src/__tests__/web/nextjs-project.test.ts:18]: Is jsiiFqn required or going to be required in the future? @eladb

**Comment by pgollucci** [comment]
well heck, next time I'll just ask and you shall receive

File: src/__tests__/web/nextjs-project.test.ts (hunk 1 above)
Comment (eladb) [reply src/__tests__/web/nextjs-project.test.ts:18]: No. It's a bug that it's included ....

Comment (Chriscbr) [reply src/__tests__/web/nextjs-project.test.ts]: I'm a little confused what your comment is referring to -- I think you're saying that jsiiFqn is only currently auto-included in projenrc files due to a bug -- so it's okay for me to remove it here? I'll go ahead and make the change.

Comment (eladb) [reply src/__tests__/web/nextjs-project.test.ts]: Exactly. Yes sorry for not being clear :-)

**Review by eladb — APPROVED** [review]

**Pull request merged**